job-processor2/src/test/java/jobprocessor2/ejb/JobStoreMessageConsumerBeanTest.java

```diff
@@ -0,0 +1,120 @@
+package jobprocessor2.ejb;
+
+import dk.dbc.commons.jsonb.JSONBContext;
+import dk.dbc.commons.jsonb.JSONBException;
+import dk.dbc.dataio.commons.types.Chunk;
+import dk.dbc.dataio.commons.types.ChunkItem;
+import dk.dbc.dataio.commons.types.ConsumedMessage;
+import dk.dbc.dataio.commons.types.Flow;
+import dk.dbc.dataio.commons.types.exceptions.InvalidMessageException;
+import dk.dbc.dataio.commons.types.jms.JmsConstants;
+import dk.dbc.dataio.commons.utils.jobstore.JobStoreServiceConnector;
+import dk.dbc.dataio.commons.utils.lang.StringUtil;
+import dk.dbc.dataio.commons.utils.test.jms.MockedJmsTextMessage;
+import dk.dbc.dataio.commons.utils.test.model.ChunkBuilder;
+import dk.dbc.dataio.commons.utils.test.model.ChunkItemBuilder;
+import dk.dbc.dataio.jobprocessor2.Metric;
+import dk.dbc.dataio.jobprocessor2.ServiceHub;
+import dk.dbc.dataio.jobprocessor2.exception.JobProcessorException;
+import dk.dbc.dataio.jobprocessor2.jms.JobStoreMessageConsumer;
+import dk.dbc.dataio.registry.PrometheusMetricRegistry;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.jms.JMSException;
+import java.util.Collections;
+import java.util.Map;
+
+import static dk.dbc.dataio.jobprocessor2.Metric.ATag.rollback;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JobStoreMessageConsumerBeanTest {
+    private static final ServiceHub SERVICE_HUB = new ServiceHub.Builder().withJobStoreServiceConnector(mock(JobStoreServiceConnector.class)).build();
+    private final Map<String, Object> headers = Map.of(
+            JmsConstants.PAYLOAD_PROPERTY_NAME, JmsConstants.CHUNK_PAYLOAD_TYPE,
+            JmsConstants.FLOW_ID_PROPERTY_NAME, 42L,
+            JmsConstants.FLOW_VERSION_PROPERTY_NAME, 1L,
+            JmsConstants.ADDITIONAL_ARGS, "{}"
+    );
+
+    @Before
+    public void init() {
+        ((PrometheusMetricRegistry)SERVICE_HUB.metricRegistry).resetAll();
+    }
+
+    @Test
+    public void onMessage_messageArgPayloadIsInvalidNewJob_noTransactionRollback() throws JMSException {
+        JobStoreMessageConsumer jobStoreMessageConsumer = new JobStoreMessageConsumer(SERVICE_HUB);
+        MockedJmsTextMessage textMessage = new MockedJmsTextMessage();
+        textMessage.setStringProperty(JmsConstants.PAYLOAD_PROPERTY_NAME, JmsConstants.CHUNK_PAYLOAD_TYPE);
+        textMessage.setText("{'invalid': 'instance'}");
+        jobStoreMessageConsumer.onMessage(textMessage);
+        Assert.assertEquals(0, Metric.dataio_message_count.counter(rollback.is("true")).getCount());
```

Comment:
Hmm er det ikke farligt at tage metrikken som indikator for om noget er gået galt/godt?

Reply:
Næh hvorfor? De skulle meget gerne tælle korrekt, og den bliver sat til 0 før hver test.